### PR TITLE
[LUM-787] Add Market Data Sync

### PR DESCRIPTION
**Pull request by Ricardo-Remy**
**Context**
We need the price history for the pools we currently have or will represent in the futur

**Implementation**
Add market data endpoint to retrieve at convenience

### CODE REVIEW

File: src/database/entities/market.entity.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { Column, CreateDateColumn, Entity, PrimaryGeneratedColumn, UpdateDateColumn } from 'typeorm';
+
+@Entity({ name: 'market_data' })
```
Comment (Segfaultd) [inline src/database/entities/market.entity.ts:3]: This shouldn't be flagged. It's not an entity, it's a JSONB structure.

File: src/http/responses/market.response.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    @Expose()
+    denom: string;
+
+    @ApiProperty()
+    @Expose()
+    price: number;
+
+    @ApiProperty()
+    @Expose()
+    created_at?: Date = null;
```
Comment (Segfaultd) [inline src/http/responses/market.response.ts:19]: `?` and `null` are redundant.

File: src/http/responses/market.response.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    @ApiProperty()
+    @Expose()
+    updated_at?: Date = null;
```
Comment (Segfaultd) [inline src/http/responses/market.response.ts:23]: `?` and `null` are redundant.

File: src/services/market.service.ts [hunk 4]
```diff
@@ -86,13 +89,42 @@ export class MarketService {
         return val.market_cap;
     };
 
-    getTokensPrices = async (symbols: string[], forceRefresh = false): Promise<{ symbol: string; price: number }[]> => {
-        if (!this._informations.length || forceRefresh) {
-            await this.refreshTokenInformations();
-        }
+    saveBulk = async (data: MarketData[]): Promise<MarketEntity> => {
```
Comment (Segfaultd) [inline src/services/market.service.ts:92]: Method is improperly named. It does not save bulk market entities. It takes a bulk of market data, then construct a new market entity and save it.

**Review by Segfaultd — CHANGES_REQUESTED** [review]

File: src/database/entities/market.entity.ts (hunk 1 above)
Comment (Ricardo-Remy) [reply src/database/entities/market.entity.ts:3]: absolutely, my bad.

File: src/database/entities/market.entity.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import { Column, CreateDateColumn, Entity, PrimaryGeneratedColumn, UpdateDateColumn } from 'typeorm';
+
+export class MarketData {
+    @Column({ type: 'varchar', length: 32 })
+    denom: string;
+
+    @Column({ type: 'float' })
```
Comment (Segfaultd) [inline src/database/entities/market.entity.ts:7]: Since it's not an entity, there is no need for `Column` decorator.

**Review by Segfaultd — APPROVED** [review]

**Pull request merged**